rotkehlchen/tasks/manager.py

```diff
@@ -210,11 +210,6 @@ def _maybe_schedule_xpub_derivation(self) -> None:
         if now - self.last_xpub_derivation_ts <= XPUB_DERIVATION_FREQUENCY:
             return
 
-        with self.database.conn.read_ctx() as cursor:
```

Comment:
I just thought that it makes really little sense to perform an extra db read since we also read from the db at the beginning of the task and nothing would break break if we don't have this check here. Moreover with this pr's changes it would require 2 reads here (both btc and bch). Or am I wrong and this is a good optimisation?

Reply:
You should keep this check.

The task should not even be launched if there is no xpubs. A DB read is nothing. Spawning a greenlet, managing and tearing it down is **a lot more expensive**.

Reply:
Got it :+1: